File: api-surface/src/services/api-security/userAccountController.ts
// @ts-ignore
/* eslint-disable */
import {request} from '@umijs/max';

/** 此处后端没有提供注释 PUT /account/api/key/status */
export async function modifyApiKeyStatus(
  body: API.ApiKeyStatusModificationDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/api/key/status`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /account/deletion */
export async function deleteAccount(
  body: API.AccountVerificationCodeCheckDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/deletion`, {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /account/new/api/key */
export async function getNewApiKey(
  body: API.AccountVerificationCodeCheckDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultString>(`/security/account/new/api/key`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 PUT /account/non/admin/authority */
export async function modifyNonAdminAuthority(
  body: API.NonAdminAuthorityModificationDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/non/admin/authority`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 PUT /account/password */
export async function modifyPassword(
  body: API.PasswordModificationDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/password`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 PUT /account/phone/or/email/binding */
export async function bindPhoneOrEmail(
  body: API.AccountVerificationCodeCheckDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/phone/or/email/binding`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /account/phone/or/email/unbinding */
export async function unbindPhoneOrEmail(
  body: API.AccountVerificationCodeCheckDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/phone/or/email/unbinding`, {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /account/registry */
export async function register(body: API.UserAccountRegistryDTO, options?: { [key: string]: any }) {
  return request<API.ResultVoid>(`/security/account/registry`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /account/secret/key */
export async function viewSecretKey(
  body: API.AccountVerificationCodeCheckDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultString>(`/security/account/secret/key`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 PUT /account/username */
export async function modifyUsername(
  body: API.UsernameModificationDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/username`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 PUT /account/username/and/password/setting */
export async function setUsernameAndPassword(
  body: API.UsernameAndPasswordSettingDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/account/username/and/password/setting`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: api-security/src/main/java/pers/project/api/security/controller/UserAccountController.java
package pers.project.api.security.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;
import pers.project.api.common.model.Result;
import pers.project.api.common.util.ResultUtils;
import pers.project.api.security.model.dto.*;
import pers.project.api.security.service.UserAccountService;

/**
 * 用户帐户控制器
 *
 * @author Luo Fei
 * @date 2023/03/22
 */
@Validated
@RestController
@RequiredArgsConstructor
@RequestMapping("/account")
public class UserAccountController {

    private final UserAccountService userAccountService;

    @PostMapping("/registry")
    public Result<Void> register(@Valid @RequestBody UserAccountRegistryDTO userAccountRegistryDTO) {
        userAccountService.createNewAccount(userAccountRegistryDTO);
        return ResultUtils.success();
    }

    @PostMapping("/new/api/key")
    public Result<String> getNewApiKey(@Valid @RequestBody AccountVerificationCodeCheckDTO codeCheckDTO) {
        String secretKey = userAccountService.generateApiKey(codeCheckDTO);
        return ResultUtils.success(secretKey);
    }

    @PostMapping("/secret/key")
    public Result<String> viewSecretKey(@Valid @RequestBody AccountVerificationCodeCheckDTO codeCheckDTO) {
        String secretKey = userAccountService.getSecretKey(codeCheckDTO);
        return ResultUtils.success(secretKey);
    }

    @PutMapping("/api/key/status")
    public Result<Void> modifyApiKeyStatus(@Valid @RequestBody ApiKeyStatusModificationDTO modificationDTO) {
        userAccountService.updateApiKeyStatus(modificationDTO);
        return ResultUtils.success();
    }

    @PutMapping("/non/admin/authority")
    public Result<Void> modifyNonAdminAuthority(@Valid @RequestBody NonAdminAuthorityModificationDTO modificationDTO) {
        userAccountService.updateNonAdminAuthority(modificationDTO);
        return ResultUtils.success();
    }

    @PutMapping("/username")
    public Result<Void> modifyUsername(@Valid @RequestBody UsernameModificationDTO modificationDTO) {
        userAccountService.updateUsername(modificationDTO);
        return ResultUtils.success();
    }

    @PutMapping("/username/and/password/setting")
    public Result<Void> setUsernameAndPassword(@Valid @RequestBody UsernameAndPasswordSettingDTO settingDTO) {
        userAccountService.updateUsernameAndPassword(settingDTO);
        return ResultUtils.success();
    }

    @PutMapping("/password")
    public Result<Void> modifyPassword(@Valid @RequestBody PasswordModificationDTO modificationDTO) {
        userAccountService.updatePassword(modificationDTO);
        return ResultUtils.success();
    }

    @DeleteMapping("/deletion")
    public Result<Void> deleteAccount(@Valid @RequestBody AccountVerificationCodeCheckDTO codeCheckDTO) {
        userAccountService.removeAccount(codeCheckDTO);
        return ResultUtils.success();
    }

    @PutMapping("/phone/or/email/binding")
    public Result<Void> bindPhoneOrEmail(@Valid @RequestBody AccountVerificationCodeCheckDTO codeCheckDTO) {
        userAccountService.savePhoneNumberOrEmailAddress(codeCheckDTO);
        return ResultUtils.success();
    }

    @DeleteMapping("/phone/or/email/unbinding")
    public Result<Void> unbindPhoneOrEmail(@Valid @RequestBody AccountVerificationCodeCheckDTO codeCheckDTO) {
        userAccountService.removePhoneNumberOrEmailAddress(codeCheckDTO);
        return ResultUtils.success();
    }

}
